src/OrleansAWSUtils/Reminders/DynamoDBReminderTable.cs

```diff
@@ -0,0 +1,294 @@
+﻿using Orleans;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Orleans.Runtime;
+using Orleans.Runtime.Configuration;
+using OrleansAWSUtils.Storage;
+using Amazon.DynamoDBv2.Model;
+using Amazon.DynamoDBv2;
+
+namespace OrleansAWSUtils.Reminders
+{
+    public class DynamoDBReminderTable : IReminderTable
+    {
+        private const string DEPLOYMENT_ID_PROPERTY_NAME = "DeploymentId";
+        private const string GRAIN_REFERENCE_PROPERTY_NAME = "GrainReference";
+        private const string REMINDER_NAME_PROPERTY_NAME = "ReminderName";
+        private const string SERVICE_ID_PROPERTY_NAME = "ServiceId";
+        private const string START_TIME_PROPERTY_NAME = "StartTime";
+        private const string PERIOD_PROPERTY_NAME = "Period";
+        private const string GRAIN_HASH_PROPERTY_NAME = "GrainHash";
+        private const string REMINDER_ID_PROPERTY_NAME = "ReminderId";
+        private const string ETAG_PROPERTY_NAME = "ETag";
+        private const string CURRENT_ETAG_ALIAS = ":currentETag";
+        private const string ETAG_INCREMENT = ":ETagIncrement";
+
+        private const string TABLE_NAME_DEFAULT_VALUE = "OrleansReminders";
+        private Logger logger;
+        private DynamoDBStorage storage;
+        private string deploymentId;
+        private Guid serviceId;
+
+        public Task Init(GlobalConfiguration config, Logger logger)
+        {
+            deploymentId = config.DeploymentId;
+            serviceId = config.ServiceId;
+
+            this.logger = logger;
+
+            storage = new DynamoDBStorage(config.DataConnectionStringForReminders, logger);
+            logger.Info(ErrorCode.MembershipBase, "Initializing AWS DynamoDB Membership Table");
```

Comment:
A new enum for all the DynamoDB-based services will be added with corresponding errors on the final PR.